Added scratchpad

niksingh710 opened the pull request:
This is a bash script that helps the user to send any focused client/window to a special:scratchpad and also makes it easier to retrieve that.

use `-h` option to get more details.

vaxerski commented:
@fufexan

fufexan commented on README.md:
```diff
@@ -16,6 +16,24 @@ Invoke shell functions in response to Hyprland socket2 events. Install with `mak
 See `shellevents_default.sh` for the supported function names. Example event files can be found in `submaps.sh` and `notifywindow.sh`.
 
+## Scratchpad
+A Bash script that instantly send focused window to a special workspace named `scratchpad`
```
```suggestion
A Bash script that instantly sends focused window to a special workspace named `scratchpad`
```

fufexan commented on README.md:
```diff
@@ -16,6 +16,24 @@ Invoke shell functions in response to Hyprland socket2 events. Install with `mak
+Usage:
+
+To send the window to scratchpad
+> bind=SUPER,s,exec,~/.config/hypr/scripts/scratchpad
+
+To get back the window
+>bind=SUPERSHIFT,s,exec,~/.config/hypr/scripts/scratchpad -g
```
You could add spacing between `=` and after `,` as it's cleaner imo.

fufexan commented on scratchpad/scratchpad:
```diff
@@ -0,0 +1,166 @@
+
+basicChecks() {
+	check hyprctl || {
+		notify "Seriously mate!!" "Use Hyprland then this script"
```
```suggestion
		notify "Seriously mate!!" "Start Hyprland before this script"
```

fufexan commented on scratchpad/scratchpad:
```diff
@@ -0,0 +1,166 @@
+		exit 1
+	}
+	pgrep -x Hyprland &>/dev/null || {
+		notify "Make Sure" "Hyprland Session is running."
```
I'm not sure having this sentence separated as title and body is ideal. Could you write it all as the title?

niksingh710 replied:
Yes, I can I guess I liked it this way because of my notification md configuration.

fufexan commented on scratchpad/scratchpad:
```diff
@@ -0,0 +1,166 @@
+        default is (${_menu_cmd[@]})
+
+  Usage:
+    bind=SUPER,s,exec,/path/to/scratchpad/binary
+    bind=SUPERSHIFT,s,exec,/path/to/scratchpad/binary -g
+
+    bind=SUPERSHIFT,s,exec,/path/to/scratchpad/binary -l -g 
```
Same comment about bind as above.

fufexan reviewed (changes requested):
Could you also add `scratchpad/scratchpad` to `.github/workflows/ci.yml` to be checked by shellcheck?

niksingh710 commented:
> Could you also add `scratchpad/scratchpad` to `.github/workflows/ci.yml` to be checked by shellcheck?

I have added that to go through from shell check.
One doubt I have.
I preferred to use `&&` and `||` shell chaining instead of `if else` to make the script a bit cleaner that is marked as an info notification from shellcheck `SC2015`
I have tested them they are not breaking anything so I also added some comments to avoid those shell check info notification in ci.

Without the comment `# shellcheck disable=SC2015` shellcheck output looks like this.
```

In scratchpad line 30:
	check notify-send && {
                          ^-- SC2015 (info): Note that A && B || C is not if-then-else. C may run when A is true.


In scratchpad line 38:
	check sed && ok "sed" || err "sed"
                  ^-- SC2015 (info): Note that A && B || C is not if-then-else. C may run when A is true.


In scratchpad line 39:
	check jq && ok "jq" || err "jq"
                 ^-- SC2015 (info): Note that A && B || C is not if-then-else. C may run when A is true.


In scratchpad line 40:
	check notify-send && ok "notify-send (Optional)" || optional "notify-send (Optional)"
                          ^-- SC2015 (info): Note that A && B || C is not if-then-else. C may run when A is true.
```

niksingh710 commented:
> Could you also add `scratchpad/scratchpad` to `.github/workflows/ci.yml` to be checked by shellcheck?

u meant `check.yml`?

fufexan commented:
> > Could you also add `scratchpad/scratchpad` to `.github/workflows/ci.yml` to be checked by shellcheck?
> 
> u meant `check.yml`?

Yes. I sometimes confuse the filenames as I have multiple repos with multiple such files.

fufexan reviewed (approved):
LGTM
